### PR TITLE
refactor: Typescript migration,  root nodes (/entry page)

### PR DESCRIPTION
**Pull request by nobeeakon**
### What
- root nodes page (`/entry`) to typescript
- split dialog content to it's own component
- UI changes in the dialog: now using material ui for input labels instead of custom ones, removed unused `select` for node type as 1) was the only option, 2) `POST` was not being affected by it.

### Screenshot
Dialog 
imagen

### CODE REVIEW

File: taxonomy-editor-frontend/src/pages/root-nodes/index.tsx [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+import Dialog from "@mui/material/Dialog";
+import DialogTitle from "@mui/material/DialogTitle";
+
+import CreateNodeDialogContent from "./CreateNodeDialogContent";
+import useFetch from "../../components/useFetch";
+import { createBaseURL } from "../editentry/createURL";
+import { toTitleCase } from "../../utils";
+import { greyHexCode } from "../../constants";
+
+type DataType = {
```
Comment (aadarsh-ram) [inline taxonomy-editor-frontend/src/pages/root-nodes/index.tsx:29]: Can we change the name of this type? Seems too abstract.

File: taxonomy-editor-frontend/src/pages/root-nodes/index.tsx [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+                <Typography variant="body1">
+                  {toTitleCase(taxonomyName ?? "")}
+                </Typography>
+              </TableCell>
+              <TableCell align="left" component="td" scope="row">
+                <Typography variant="body1">{branchName}</Typography>
+              </TableCell>
+            </TableBody>
+          </Table>
+        </TableContainer>
```
Comment (aadarsh-ram) [inline taxonomy-editor-frontend/src/pages/root-nodes/index.tsx:120]: If possible, can we add a bottom border for this table component? It seems out of place without it.

File: taxonomy-editor-frontend/src/pages/root-nodes/index.tsx (hunk 1 above)
Comment (nobeeakon) [reply taxonomy-editor-frontend/src/pages/root-nodes/index.tsx:29]: done, added backend type

**Review by aadarsh-ram — APPROVED** [review]
LGTM, thanks! :)

**Pull request merged**